test/test_run_analysis.rb

```diff
@@ -170,6 +170,20 @@ def test_building_id
     assert(!File.exist?(File.join(@testing_baseline, 'run2')))
   end
 
+  def test_upgrade_name
```

Comment:
Can you also add a test where someone uses an invalid `upgrade_name`? Ideally it would catch and throw an error right away (not after reading TSV, sampling, etc.), since all it needs against is the yaml.

Reply:
I added `test_errors_invalid_upgrade_name`.

Reply:
I guess it's better than nothing, but it would have been helpful to tell the user which upgrade names are invalid. And what the list of valid upgrade names are.